### PR TITLE
Upgrade to Go 1.18

### PR DESCRIPTION
**Pull request by epk**
### Description of your changes

- Upgrades Go to 1.18
- Fixes some minor lint errors
- Update `github.com/99designs/gqlgen` to latest to support Go 1.18

I have:

- [x] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR, as appropriate.

### CODE REVIEW

**Review by tnthornton — APPROVED** [review]
LGTM. Thanks @epk !

File: internal/graph/model/id.go [hunk 1]
```diff
@@ -19,7 +19,7 @@ import (
 	"io"
 	"strings"
 
-	"github.com/kjk/smaz"
+	"github.com/epk/smaz"
```
Comment (epk) [inline internal/graph/model/id.go:22]: https://github.com/kjk/smaz does not have a `go.mod` and has a relative import which is no longer allowed  in Go 1.18

Diff here: https://github.com/epk/smaz/commit/c11a89997fcfe65ded3d3bc79694ce50e675f766

Comment (negz) [reply internal/graph/model/id.go:22]: https://github.com/upbound/xgql/blob/main/internal/graph/model/id.go#L115

Note that updates to smaz logic could result in a breaking change for xgql per the comment above. That said, at a glance this change seems safe to me.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -10,9 +10,9 @@ on:
 
 env:
   # Common versions
-  GO_VERSION: '1.16'
-  GOLANGCI_VERSION: 'v1.31'
-  DOCKER_BUILDX_VERSION: 'v0.4.2'
+  GO_VERSION: '1.18'
+  GOLANGCI_VERSION: 'v1.47.1'
+  DOCKER_BUILDX_VERSION: 'v0.8.2'
```
Comment (negz) [inline .github/workflows/ci.yml:15]: We copy and paste this CI flow around to a lot of places. Ideally we'd update them all but I don't want to volunteer you to update the world - maybe a tracking issue would be appropriate?

Comment (epk) [reply .github/workflows/ci.yml:15]: I'll open a tracking issue in one of the internal repos and maybe write a one-off script to automate these PRs

File: internal/graph/model/id.go (hunk 1 above)
Comment (epk) [reply internal/graph/model/id.go:22]: Yeah, I only added a `go.mod`, removed relative imports and ran the code through https://github.com/mvdan/gofumpt

**Pull request merged**